File: ShowBatteryStatus/app/src/main/java/com/blackbirds/shakil/showbatterystatus/MainActivity.java
package com.blackbirds.shakil.showbatterystatus;

import androidx.appcompat.app.AppCompatActivity;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.BatteryManager;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    TextView txtBatteryStatus;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        txtBatteryStatus = findViewById(R.id.txtBatteryStatus);

        registerReceiver(this.mBatteryInfoReceiver, new IntentFilter(Intent.ACTION_BATTERY_CHANGED));
    }

    private BroadcastReceiver mBatteryInfoReceiver = new BroadcastReceiver() {
        @Override
        public void onReceive(Context context, Intent intent) {
            int level = intent.getIntExtra(BatteryManager.EXTRA_LEVEL, -1);
            int scale = intent.getIntExtra(BatteryManager.EXTRA_SCALE, -1);

            float batteryPct = level * 100 / (float)scale;

            txtBatteryStatus.setText("Battery Status: " + batteryPct + "%");
        }
    };
}

File: KotlinShowBatteryStatus/app/src/main/java/com/blackbirds/shakil/kotlinshowbatterystatus/MainActivity.kt
package com.blackbirds.shakil.kotlinshowbatterystatus

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.os.BatteryManager
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    var txtBatteryStatus: TextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        txtBatteryStatus = findViewById(R.id.txtBatteryStatus)

        registerReceiver(this.mBatteryInfoReceiver, IntentFilter(Intent.ACTION_BATTERY_CHANGED))
    }

    private val mBatteryInfoReceiver: BroadcastReceiver = object : BroadcastReceiver() {
        override fun onReceive(context: Context?, intent: Intent?) {
            val level = intent!!.getIntExtra(BatteryManager.EXTRA_LEVEL,-1)
            val scale = intent!!.getIntExtra(BatteryManager.EXTRA_SCALE,-1)

            val batteryPct = level * 100 / scale.toFloat()

            txtBatteryStatus!!.text = "Battery Status: $batteryPct%"
        }
    }
}